### PR TITLE
Kaisa/enhancement/278 update gallery to work with cms integration

### LINKED ISSUE #278: Update Gallery to Work with CMS Integration
### Description
We need to update the galleries to function through the CMS for better content management and optimization.

### ToDo List:
1. **Photo Categorization**: Photos should be organized by categories within the CMS.  
   - **Consider creating categories as a separate entity**

2. **Photo Object Structure**: Each photo should be represented as an object containing:
   - **Preview and Full Versions**: Each version should specify image sizes to leverage Next.js image optimization.
   - **Multilingual Alt Text**: Include alt text for each language (check with Joni for details on i18n configuration).

3. **RTK Query and Entities**: Implement the gallery feature using entities and RTK Query, similar to the setup for members, to streamline data fetching and state management. (I think you could create two apis within Gallery entity, first for Gallery second for GalleryCategories)

4. **Frontend Dropdown Widget**:
   - Add a dropdown widget on the frontend for easier navigation, similar to [this example on mobile](https://altzone.fi/fi/heroes/ahmat).
   - For the desktop version, plan to implement this using the `shared/ui/layout/layoutWithSidebars` component (currently in development but can be used in this initial phase).

### Example Structure
The following is an example of how the data structure for photos in the CMS could look in TypeScript:

```typescript
// Example of a photo object structure in TypeScript for CMS integration

interface PhotoVersion {
  url: string;
  width: number;
  height: number;
  altText: {
    [languageCode: string]: string; // Multilingual alt text by language code, e.g., { "en": "Sample alt text", "fi": "Esimerkki vaihtoehtoinen teksti" }
  };
}

interface PhotoObject {
  id: string;
  categoryId: string; // Reference to a separate category entity for better translation support
  versions: {
    preview: PhotoVersion;
    full: PhotoVersion;
  };
}

interface Category {
  id: string;
  name: {
    [languageCode: string]: string; // Multilingual category name, e.g., { "en": "Nature", "fi": "Luonto" }
  };
}

// Sample data
const galleryData: PhotoObject[] = [
  {
    id: "photo1",
    categoryId: "category1",
    versions: {
      preview: {
        url: "https://example.com/photos/photo1_preview.jpg",
        width: 300,
        height: 200,
        altText: {
          en: "A beautiful forest",
          fi: "Kaunis metsä",
        },
      },
      full: {
        url: "https://example.com/photos/photo1_full.jpg",
        width: 1920,
        height: 1080,
        altText: {
          en: "A beautiful forest in high resolution",
          fi: "Kaunis metsä korkearesoluutiolla",
        },
      },
    },
  },
];

const categories: Category[] = [
  {
    id: "category1",
    name: {
      en: "Nature",
      fi: "Luonto",
    },
  },
  {
    id: "category2",
    name: {
      en: "Urban",
      fi: "Kaupunki",
    },
  },
];

### PR DESCRIPTION
**Pull request by kaisahakola**
## 📄 **Pull Request Overview**

closes #278 

## 🔧 **Changes Made**

1. Replaced gallery's old API logic with Directus.

2. Updated gallery components to work with new API.
These components include:
- ImageWall
- SectionGallery (version 2)
- PictureGalleryPage

---

## ✅ **Checklist Before Submission**

- **Functionality**: I have tested my code, and it works as expected.
- **JSDoc**: I have added or updated JSDoc comments for all relevant code.
- **Debugging**: No `console.log()` or other debugging statements are left.
- **Clean code**: Removed commented-out or unnecessary code.
- **Tests**: Added new tests or updated existing ones for the changes made. 
---

### CODE REVIEW

**Comment by Skoivumaki** [comment]
Build fails to compile. Fix it first and make sure both build and tests pass.

File: frontend-next-migration/src/entities/Gallery/api/useGetDirectusGalleryImages.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import { envHelper } from '@/shared/const/envHelper';
+import { useMemo } from 'react';
+import {
+    useGetGalleryCategoriesQuery,
+    useGetPhotoObjectsQuery,
+    useGetPhotoVersionsQuery,
+    getPhotoVersionTranslation,
+    getCategoryTranslation,
```
Comment (leolabdev) [inline frontend-next-migration/src/entities/Gallery/api/useGetDirectusGalleryImages.ts:8]: please use relative paths when work inside the same module

File: frontend-next-migration/src/entities/Gallery/api/useGetDirectusGalleryImages.test.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
+import { renderHook } from '@testing-library/react';
+import { useGetDirectusGalleryImages } from '../api/useGetDirectusGalleryImages';
+import {
```
Comment (leolabdev) [inline frontend-next-migration/src/entities/Gallery/api/useGetDirectusGalleryImages.test.ts:3]: please use relative paths when work inside the same module

File: frontend-next-migration/src/entities/Gallery/api/translations.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { PhotoVersionTranslations, CategoryTranslations } from '@/entities/Gallery';
```
Comment (leolabdev) [inline frontend-next-migration/src/entities/Gallery/api/translations.ts:1]: please use relative paths when work inside the same module

File: frontend-next-migration/src/entities/Gallery/api/filterAndTransformImages.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { PhotoObject, PhotoVersion } from '@/entities/Gallery';
```
Comment (leolabdev) [inline frontend-next-migration/src/entities/Gallery/api/filterAndTransformImages.ts:1]: please use relative paths when work inside the same module

File: frontend-next-migration/src/entities/Gallery/ui/ImageWall/ImageWall.stories.tsx [hunk 5]
```diff
@@ -1,6 +1,5 @@
 import { StoryObj } from '@storybook/react';
-import { ImageWall } from './ImageWall';
-import { ImageData } from '@/entities/Gallery';
+import { mockImagesFull, mockImagesPreview, ImageWall } from '@/entities/Gallery';
```
Comment (leolabdev) [inline frontend-next-migration/src/entities/Gallery/ui/ImageWall/ImageWall.stories.tsx:2]: please use relative paths when work inside the same module

File: frontend-next-migration/src/entities/Gallery/ui/ImageWall/ImageWall.tsx [hunk 6]
```diff
@@ -1,32 +1,41 @@
 'use client';
 import cls from './ImageWall.module.scss';
 import Image from 'next/image';
-import { ImageData } from '../../types/gallery';
+import { PhotoObject, PhotoVersion, filterAndTransformImages } from '@/entities/Gallery';
```
Comment (leolabdev) [inline frontend-next-migration/src/entities/Gallery/ui/ImageWall/ImageWall.tsx:4]: please use relative paths when work inside the same module

File: frontend-next-migration/src/features/NavigateGalleries/ui/GalleryNavMenuAsDropdown.tsx [hunk 7]
```diff
@@ -1,27 +1,35 @@
...
+import { useParams } from 'next/navigation';
+import { getRouteGalleryCategoryPage } from '@/shared/appLinks/RoutePaths';
 
 interface GalleryNavMenuProps {
     // className?: string;
     openByDefault?: boolean;
 }
 
-const categories: string[] = ['All', 'Nature', 'City', 'People', 'Technology'];
+const categoriesEn: string[] = ['All', 'Nature', 'City', 'People', 'Technology'];
```
Comment (leolabdev) [inline frontend-next-migration/src/features/NavigateGalleries/ui/GalleryNavMenuAsDropdown.tsx:15]: use api values instead

**Review by leolabdev — CHANGES_REQUESTED** [review]

**Review by leolabdev — APPROVED** [review]

**Pull request merged**